src/ophyd_async/core/signal.py

```diff
@@ -253,6 +257,52 @@ def set_sim_callback(signal: Signal[T], callback: ReadingValueCallback[T]) -> No
     return _sim_backends[signal].set_callback(callback)
 
 
+async def verify_readable(
+    func: Callable[[], dict[str, Any] | Awaitable[dict[str, Any]]],
+    expectation: Dict[str, Any],
+) -> None:
+    """Loop thorough dictionary and verify it against
+    a expected dictionary (readable)"""
+    if asyncio.iscoroutinefunction(func):
+        result = await func()
+    else:
+        result = func()
+
+    for signal in expectation:
+        for field in expectation[signal]:
+            if field == "timestamp":
+                assert isinstance(result[signal]["timestamp"], float)
+            else:
+                assert result[signal][field] == expectation[signal][field]
+
+
+async def assert_value(signal: SignalR[T], value: T) -> None:
+    """Assert a signal value"""
+    assert await signal.get_value() == value
+
+
+async def assert_reading(
+    readable: Readable, reading: Dict[str, Reading] | dict[str, dict[str, Any]]
+) -> None:
+    """Assert readable by calling verify_readable"""
+    await verify_readable(readable.read, reading)
+
+
+async def assert_configuration(
+    configurable: Configurable,
+    configuration: Dict[str, Reading] | dict[str, dict[str, Any]],
+) -> None:
+    """Assert readable generated by configurable.read_configuration
+    using verify_readable"""
+    await verify_readable(configurable.read_configuration, configuration)
+
+
+def assert_emitted(docs: Dict[str, list], **numbers: int):
```

Comment:
You can import `DocumentType` from event model

```suggestion
def assert_emitted(docs: DocumentType, **numbers: int):
```

Reply:
Although docs is a dict of document name to list of document instances, so I think that should be:
```suggestion
def assert_emitted(docs: Dict[str, list[DocumentType]], **numbers: int):
```
Except `DocumentType` is actually the types, which is not what we want, we would quite like a similar `Document` union in event model...

Reply:
so in the absence of that it's probably as correct as it could be...